client/components/App.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+	<div
+		id="viewport"
+		role="tablist">
+		<aside id="sidebar">
+			<div class="scrollable-area">
+				<div class="logo-container">
+					<img
+						:src="'img/logo-' + ( isPublic() ? 'horizontal-' : '' ) + 'transparent-bg.svg'"
+						class="logo"
+						alt="The Lounge">
+					<img
+						:src="'img/logo-' + ( isPublic() ? 'horizontal-' : '' ) + 'transparent-bg-inverted.svg'"
+						class="logo-inverted"
+						alt="The Lounge">
+				</div>
+				<NetworkList
+					:networks="networks"
+					:active-channel="activeChannel" />
+			</div>
+			<footer id="footer">
+				<span
+					class="tooltipped tooltipped-n tooltipped-no-touch"
+					aria-label="Sign in"><button
+						class="icon sign-in"
+						data-target="#sign-in"
+						aria-label="Sign in"
+						role="tab"
+						aria-controls="sign-in"
+						aria-selected="false" /></span>
+				<span
+					class="tooltipped tooltipped-n tooltipped-no-touch"
+					aria-label="Connect to network"><button
+						class="icon connect"
+						data-target="#connect"
+						aria-label="Connect to network"
+						role="tab"
+						aria-controls="connect"
+						aria-selected="false" /></span>
+				<span
+					class="tooltipped tooltipped-n tooltipped-no-touch"
+					aria-label="Settings"><button
+						class="icon settings"
+						data-target="#settings"
+						aria-label="Settings"
+						role="tab"
+						aria-controls="settings"
+						aria-selected="false" /></span>
+				<span
+					class="tooltipped tooltipped-n tooltipped-no-touch"
+					aria-label="Help"><button
+						class="icon help"
+						data-target="#help"
+						aria-label="Help"
+						role="tab"
+						aria-controls="help"
+						aria-selected="false" /></span>
+			</footer>
+		</aside>
+		<div id="sidebar-overlay" />
+		<article id="windows">
+			<Chat
+				v-if="activeChannel"
+				:network="activeChannel.network"
+				:channel="activeChannel.channel" />
+			<div
+				id="sign-in"
+				class="window"
+				role="tabpanel"
+				aria-label="Sign-in" />
+			<div
+				id="connect"
+				class="window"
+				role="tabpanel"
+				aria-label="Connect" />
+			<div
+				id="settings"
+				class="window"
+				role="tabpanel"
+				aria-label="Settings" />
+			<div
+				id="help"
+				class="window"
+				role="tabpanel"
+				aria-label="Help" />
+			<div
+				id="changelog"
+				class="window"
+				aria-label="Changelog" />
+		</article>
+	</div>
+</template>
+
+<script>
+const throttle = require("lodash/throttle");
+
+import NetworkList from "./NetworkList.vue";
+import Chat from "./Chat.vue";
+
+export default {
```

Comment:
Sidenote, but at work we found ourselves almost exclusively using named imports. Reasons are multiple, but:

- TS historically used to be a pain on default imports (doesn't apply here)
- That kinda enforces naming being used across app/app (`import { App } from '../App' and, yes, I know you can rename, but it has to be explicit to override as opposed to typo). This makes searching across codebase very nice.
- You can't `export default` an ES6 class
- It makes for nice index files where we import/export everything in our libraries: `export { ThisThing } from './src/ThisThing'` (as opposed to `export default as ThisThing from '...'` or whatever the syntax is).

Not asking for any change here, just meant to share experience :)

Reply:
Looks like what you are suggesting is currently experimental: https://github.com/vuejs/vue-class-component